tests/e2e/scale/test_pods_are_not_oomkilled_while_running_ios.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.framework.testlib import E2ETest, scale, tier4
+from tests.helpers import (
+    default_storage_class,
+    validate_pods_are_running_and_not_restarted,
+    validate_pod_oomkilled
+)
+from ocs_ci.utility.utils import ceph_health_check
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@scale
+@pytest.mark.parametrize(
+    argnames=["interface"],
+    argvalues=[
+        pytest.param(
+            constants.CEPHBLOCKPOOL, marks=pytest.mark.polarion_id("OCS-2048")
+        ),
+        pytest.param(
+            constants.CEPHFILESYSTEM, marks=pytest.mark.polarion_id("OCS-2049")
+        )
+    ]
+)
+class TestPodAreNotOomkilledWhileRunningIO(E2ETest):
+    """
+    A test case to validate pods are in running
+    state when IOs are running continuously
+
+    """
+    pvc_size_gb = 2048
+    io_size = 1024
+
+    @pytest.fixture()
+    def base_setup(
+        self, interface, pvc_factory, dc_pod_factory
+    ):
+        """
+        A setup phase for the test:
+        get all the ceph pods information,
+        create maxsize pvc, pod and run IO
+
+        """
+
+        self.pod_obj = get_all_pods(
```

Comment:
I suggest not to use classes with `self` references later in test code. It would be better to just return the created objects and access them directly:
```
return (pod_obj, sc, pvc_obj, pod_obj, dc_pod_obj)
```
Or just pod_obj as you don't use other objects in the test:
```
return pod_obj
```
Also I would rename pod_obj to pod_objs so that it's clear that there are multiple objects in the variable.

Reply:
Thanks Filip, changes made accordingly